src/main/java/duke/GuiParser.java

```diff
@@ -0,0 +1,163 @@
+package duke;
+
+import java.io.FileNotFoundException;
+import java.util.ArrayList;
+import java.util.Arrays;
+
+/**
+ * Represents a parser object that interprets user inputs.
+ */
+public class GuiParser {
+
+    public GuiParser() {}
+
+    /**
+     * Interprets the given user input with the current task list and storage.
+     *
+     * @param input Current input string of the user.
+     * @param taskList Current list of tasks of the user.
+     * @param storage Current storage of the user.
+     * @throws duke.DukeException
+     * @throws FileNotFoundException
+     */
+    public String interpretGui(String input, duke.TaskList taskList, duke.Storage storage)
```

Comment:
may want to import duke.TaskList above so that you can just refer to it as TaskList, not a major thing though!:)